python/cugraph/cugraph/tests/test_sorensen.py

```diff
@@ -0,0 +1,239 @@
+        edges.append((destinations[i], sources[i]))
+    edges = list(dict.fromkeys(edges))
+    edges = sorted(edges)
+    # in NVGRAPH tests we read as CSR and feed as CSC, so here we doing this
+    # explicitly
+    print("Format conversion ... ")
+
+    Gnx = nx.from_pandas_edgelist(
+        M, source="0", target="1", edge_attr="weight", create_using=nx.Graph()
+    )
+
+    # Networkx Jaccard Call
+    print("Solving... ")
+    t1 = time.time()
+    preds = nx.jaccard_coefficient(Gnx, edges)
+    t2 = time.time() - t1
+
+    print("Time : " + str(t2))
+    src = []
+    dst = []
+    coeff = []
+    for u, v, p in preds:
+        src.append(u)
+        dst.append(v)
+        # Conversion from Networkx Jaccard to Sorensen
+        # No networkX equivalent
+        coeff.append((2*p)/(1+p))
+    return src, dst, coeff
+
+
+@pytest.mark.parametrize("graph_file", utils.DATASETS_UNDIRECTED)
+def test_sorensen(graph_file):
+    gc.collect()
+
+    M = utils.read_csv_for_nx(graph_file)
+    cu_M = utils.read_csv_file(graph_file)
+    cu_src, cu_dst, cu_coeff = cugraph_call(cu_M)
+    nx_src, nx_dst, nx_coeff = networkx_call(M)
+
+    # Calculating mismatch
+    err = 0
+    tol = 1.0e-06
+
+    assert len(cu_coeff) == len(nx_coeff)
+    for i in range(len(cu_coeff)):
+        if abs(cu_coeff[i] - nx_coeff[i]) > tol * 1.1:
+            err += 1
+
+    print("Mismatches:  %d" % err)
+    assert err == 0
+
+
+@pytest.mark.parametrize("graph_file", [PurePath(
+    utils.RAPIDS_DATASET_ROOT_DIR)/"netscience.csv"]
+)
+def test_sorensen_edgevals(graph_file):
+    gc.collect()
+
+    M = utils.read_csv_for_nx(graph_file)
+    cu_M = utils.read_csv_file(graph_file)
+    cu_src, cu_dst, cu_coeff = cugraph_call(cu_M, edgevals=True)
+    nx_src, nx_dst, nx_coeff = networkx_call(M)
+
+    # Calculating mismatch
+    err = 0
+    tol = 1.0e-06
+
+    assert len(cu_coeff) == len(nx_coeff)
+    for i in range(len(cu_coeff)):
+        if abs(cu_coeff[i] - nx_coeff[i]) > tol * 1.1:
+            err += 1
+
+    print("Mismatches:  %d" % err)
+    assert err == 0
+
+
+@pytest.mark.parametrize("graph_file", utils.DATASETS_UNDIRECTED)
+def test_sorensen_two_hop(graph_file):
+    gc.collect()
+
+    M = utils.read_csv_for_nx(graph_file)
+    cu_M = utils.read_csv_file(graph_file)
+
+    Gnx = nx.from_pandas_edgelist(
+        M, source="0", target="1", create_using=nx.Graph()
+    )
+    G = cugraph.Graph()
+    G.from_cudf_edgelist(cu_M, source="0", destination="1")
+    pairs = (
+        G.get_two_hop_neighbors()
+        .sort_values(["first", "second"])
+        .reset_index(drop=True)
+    )
+    nx_pairs = []
+    for i in range(len(pairs)):
+        nx_pairs.append((pairs["first"].iloc[i], pairs["second"].iloc[i]))
+    preds = nx.jaccard_coefficient(Gnx, nx_pairs)
+    nx_coeff = []
+    for u, v, p in preds:
+        # Conversion from Networkx Jaccard to Sorensen
+        # No networkX equivalent
+        nx_coeff.append((2*p)/(1+p))
```

Comment:
same comment as above: seems like we'd be better off comparing to hard-coded good results instead of recomputing each time.

Reply:
Added a FIXME for that